more_itertools/more.py

```diff
@@ -1527,6 +1528,24 @@ def split_into(iterable, sizes):
             yield list(islice(it, size))
 
 
+def chain_if(*iterables, predicate=lambda x, y: True, test_first=False):
+    """
+    Join iterables when predicate is True. If no predicate is given,
+    join all iterables. Does not modify the original iterables. Does not
+    work with infinite iterables. Joins iterables in order they are given.
+    >>> list(chain_if([1], [3], [2], predicate=lambda x,y: x[0]+y[0] == 3))
+    [1, 2]
+    """
+    if not iterables:
```

Comment:
Why this special case code? You get the same result without it, and it costs extra time for every normal case.

Reply:
You are absolutely right. Removed.